airflow/jobs/scheduler_job.py

```diff
@@ -189,80 +189,6 @@ def is_alive(self, grace_multiplier: Optional[float] = None) -> bool:
             and (timezone.utcnow() - self.latest_heartbeat).total_seconds() < scheduler_health_check_threshold
         )
 
-    @provide_session
-    def _change_state_for_tis_without_dagrun(
```

Comment:
This is deleted, as the DB won't let us have TIs without Dagruns anymore :)

Reply:
Huge, can clear the issue of tasks getting stuck!

Reply:
Since tasks will no longer exist without dagrun, obviously the dag will fail or succeed and tis won't be stuck. I think we should remove these lines.:
https://github.com/apache/airflow/blob/5d90bcb93a4318de7db7fd5527bb45c823fb0dab/airflow/jobs/scheduler_job.py#L603-L604

Reply:
I think this can happen if (for example) there is a crash in the worker early on in the "boot" process, so it can't report it's own status in the DB.